File: slp/backend-dotnet/Features/Dashboard/DashboardController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using backend_dotnet.Features.Dashboard;
using System.Security.Claims;

namespace backend_dotnet.Features.Dashboard;

[ApiController]
[Route("api/[controller]")]
[Authorize]
public class DashboardController : ControllerBase
{
    private readonly IDashboardService _dashboardService;

    public DashboardController(IDashboardService dashboardService)
    {
        _dashboardService = dashboardService;
    }

    [HttpGet("word-of-the-day")]
    public async Task<ActionResult<WordOfTheDayDto>> GetWordOfTheDay()
    {
        var word = await _dashboardService.GetWordOfTheDayAsync();
        return Ok(word);
    }

    [HttpGet("top-quizzes")]
    public async Task<ActionResult<List<TopQuizDto>>> GetTopQuizzes([FromQuery] int limit = 5)
    {
        if (limit < 1 || limit > 20) limit = 5;
        var quizzes = await _dashboardService.GetTopQuizzesAsync(limit);
        return Ok(quizzes);
    }

    [HttpGet("user-stats")]
    public async Task<ActionResult<UserStatsDto>> GetUserStats()
    {
        var userIdClaim = User.FindFirst(ClaimTypes.NameIdentifier)?.Value;
        if (!int.TryParse(userIdClaim, out var userId))
            return Unauthorized();

        var stats = await _dashboardService.GetUserStatsAsync(userId);
        return Ok(stats);
    }
}

File: slp/backend-nestjs/src/modules/dashboard/dashboard.controller.ts
import { Controller, Get, Query, UseGuards, Request } from '@nestjs/common';
import { SessionGuard } from '../session/session.guard';
import { DashboardService } from './dashboard.service';
import { TopQuizDto } from './dto/top-quiz.dto';
import { UserStatsDto } from './dto/user-stats.dto';
import { WordOfTheDayDto } from './dto/word-of-the-day.dto';

@Controller('api/dashboard')
@UseGuards(SessionGuard)
export class DashboardController {
  constructor(private readonly dashboardService: DashboardService) {}

  @Get('word-of-the-day')
  async getWordOfTheDay(): Promise<WordOfTheDayDto> {
    return this.dashboardService.getWordOfTheDay();
  }

  @Get('top-quizzes')
  async getTopQuizzes(@Query('limit') limitStr?: string): Promise<TopQuizDto[]> {
    let limit = limitStr ? parseInt(limitStr, 10) : 5;
    if (isNaN(limit) || limit < 1) limit = 5;
    if (limit > 20) limit = 20;
    return this.dashboardService.getTopQuizzes(limit);
  }

  @Get('user-stats')
  async getUserStats(@Request() req): Promise<UserStatsDto> {
    const userId = req.user.id;
    return this.dashboardService.getUserStats(userId);
  }
}